packages/components/src/components/navigation-item/navigation-item.test.ts

```diff
@@ -0,0 +1,313 @@
+import type SdNavigationItem from './navigation-item';
+
+const defaultSlot = 'Default Slot';
+const childrenSlot = html`<div slot="children">Children</div>`;
+const variants = {
+  button: {
+    default: html`<sd-navigation-item>${defaultSlot}</sd-navigation-item>`,
+    current: html`<sd-navigation-item current>${defaultSlot}</sd-navigation-item>`,
+    disabled: html`<sd-navigation-item disabled>${defaultSlot}</sd-navigation-item>`
+  },
+  link: {
+    default: html`<sd-navigation-item href="#">${defaultSlot}</sd-navigation-item>`,
+    current: html`<sd-navigation-item current href="#">${defaultSlot}</sd-navigation-item>`,
+    disabled: html`<sd-navigation-item disabled href="#">${defaultSlot}</sd-navigation-item>`,
+    children: html`<sd-navigation-item href="#">${defaultSlot}${childrenSlot}</sd-navigation-item>`
+  },
+  accordion: {
+    default: html`<sd-navigation-item>${defaultSlot}${childrenSlot}</sd-navigation-item>`,
+    current: html`<sd-navigation-item current>${defaultSlot}${childrenSlot}</sd-navigation-item>`,
+    disabled: html`<sd-navigation-item disabled>${defaultSlot}${childrenSlot}</sd-navigation-item>`
+  }
+};
+
+describe('<sd-navigation-item>', () => {
+  // Test default button variant
+  describe('by default', () => {
+    // Render
+    it('only renders a button element', async () => {
+      const el = await fixture<SdNavigationItem>(variants.button.default);
+
+      expect(el).to.exist;
+      expect(el.shadowRoot!.querySelector('a')).to.not.exist;
+      expect(el.shadowRoot!.querySelector('button')).to.exist;
+      expect(el.shadowRoot!.querySelector('summary')).to.not.exist;
+      expect(el.shadowRoot!.querySelector('details')).to.not.exist;
+    });
+
+    // Accessibility
+    it('passes accessibility test', async () => {
+      const el = await fixture<SdNavigationItem>(variants.button.default);
+      await expect(el).to.be.accessible();
+    });
+
+    it('adds aria-current: "page" to <a> when current is true', async () => {
+      const el = await fixture<SdNavigationItem>(variants.link.current);
+      const link = el.shadowRoot!.querySelector('a');
+      expect(link).attribute('aria-current', 'page');
+    });
+
+    // Events
+    describe('when disabled', () => {
+      it('passes accessibility test', async () => {
+        const el = await fixture<SdNavigationItem>(variants.button.disabled);
+        await expect(el).to.be.accessible();
+      });
+
+      it('adds aria-disabled', async () => {
+        const el = await fixture<SdNavigationItem>(variants.button.disabled);
+        const button = el.shadowRoot!.querySelector('button');
+        expect(button).attribute('aria-disabled');
+      });
+
+      it('should disable the native <button>', async () => {
+        const el = await fixture<SdNavigationItem>(variants.button.disabled);
+        expect(el.shadowRoot!.querySelector('button[disabled]')).to.exist;
+      });
+
+      // it('should not emit any events', async () => {
```

Comment:
pls remove comments here and in case your not able to fix the tests, pls open up an issue to add the test back to the component. Goal is to have clean test files as disabled tests usually get lost after a while... ;-)

Reply:
Of course, I was hoping to get feedback during the review and act accordingly...